presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -67,11 +71,26 @@ def anonymize(
             anonymized_text = self.__extract_anonymizer_and_anonymize(
                 analyzer_result.entity_type, anonymizer_config, text_to_anonymize
             )
-            text_builder.replace_text(
+            index_from_end = text_builder.replace_text(
                 anonymized_text, analyzer_result.start, analyzer_result.end
             )
 
-        return text_builder.output_text
+            result_item = AnonymizerResultItem(anonymizer_config.anonymizer_name,
+                                               analyzer_result.entity_type,
+                                               0,
+                                               index_from_end,
+                                               anonymized_text)
+
+            anonymizer_result.add_item(result_item)
+
+        # normalize the indexes to be index from start
+        text_len = len(text_builder.output_text)
```

Comment:
Please consider moving this logic to another method. 
It can even be a AnonymizerResult method which gets the params and know how to normalize them.